optimum/amd/brevitas/quantizer.py

```diff
@@ -0,0 +1,327 @@
+                f"Applying Activation Equalization {quantization_config.activations_equalization} (SmoothQuant)..."
+            )
+            apply_act_equalization(model, quantization_config.activations_equalization, calibration_dataset)
+            logger.info("Activation equalization applied.")
+
+        # We do not quantize embedding and last fully connected layer
+        model = quantize_model(
+            model,
+            dtype=dtype,
+            weight_quant_format="int",
+            weight_quant_type="sym" if quantization_config.weights_symmetric else "asym",
+            weight_bit_width=quantization_config.weights_bitwidth,
+            weight_param_method=quantization_config.weights_param_method,
+            weight_scale_precision=quantization_config.scale_precision,
+            weight_quant_granularity=quantization_config.weights_quant_granularity,
+            weight_group_size=quantization_config.weights_group_size,
+            quantize_weight_zero_point=quantization_config.quantize_zero_point,
+            input_bit_width=quantization_config.activations_bitwidth,
+            input_quant_type="sym" if quantization_config.activations_symmetric else "asym",
+            input_quant_format="int",
+            input_param_method=quantization_config.activations_param_method,
+            input_scale_precision=quantization_config.scale_precision,
+            input_scale_type="static" if quantization_config.is_static else "dynamic",
+            input_quant_granularity=quantization_config.activations_quant_granularity,
+            input_group_size=quantization_config.activations_group_size,
+            quantize_input_zero_point=quantization_config.quantize_zero_point,
+        )
+
+        # Perform a single inference pass to generate the correct state_dict
+        if quantization_config.apply_gptq or quantization_config.is_static:
+            with torch.no_grad():
+                model(**calibration_dataset[0])
+
+        if quantization_config.apply_gptq:
+            logger.info("Applying gptq...")
+            apply_gptq(
+                model,
+                calibration_dataset,
+                act_order=quantization_config.gptq_act_oder,
+                group_of_parallel_layers=self.group_of_parallel_layers,
+            )
+            logger.info("GPTQ applied.")
+
+        if quantization_config.activations_bitwidth is not None and quantization_config.is_static:
+            logger.info("Applying activation calibration...")
+            apply_calibration(model, calibration_dataset)
+            logger.info("Activation calibration applied.")
+
+        if quantization_config.apply_bias_correction:
+            logger.info("Applying Bias Correction...")
+            apply_bias_correction(
+                model,
+                calibration_dataset,
+            )
+            logger.info("Bias Correction applied.")
+
+        return model
+
+    """
+    TODO: test this, and maybe use it by default?
```

Comment:
We can use by default, however it requires the user to specify the names of the layers.
I am not sure there is a way to do this automatically.

Reply:
Let's leave it for a later PR?

Reply:
Fine for me